javatests/arcs/core/storage/ReferenceModeStoreTest.kt

```diff
@@ -674,6 +674,30 @@ class ReferenceModeStoreTest {
         job.join()
     }
 
+    @Test
+    fun backingStoresCleanedUpWhenLastCallbackRemoved() = runBlocking {
```

Comment:
Might be good to have a test that verifies the behavior of the RefModeStore when the last callback was removed, then a new callback was added. (Going from non-empty to empty and back to non-empty)

Reply:
Such a test will fail with the current implementation. I would like to address this in a different PR. One proposed solution is here: #5712

Reply:
(The onUnbind PR was another option, but Filippo pointed out that it won't work, so there's more to be done there to make that a viable option as well)